Mathlib/NumberTheory/NumberField/Discriminant.lean

```diff
@@ -80,6 +90,108 @@ theorem _root_.NumberField.mixedEmbedding.volume_fundamentalDomain_latticeBasis
+    ∃ (a : 𝓞 K), a ≠ 0 ∧
+      |Algebra.norm ℚ (a:K)| ≤ (4 / π) ^ NrComplexPlaces K *
+        (finrank ℚ K).factorial / (finrank ℚ K) ^ (finrank ℚ K) * Real.sqrt |discr K| := by
+  -- The smallest possible value for `exists_ne_zero_mem_ringOfIntegers_of_norm_le`
+  let B := (minkowskiBound K * (convexBodySumFactor K)⁻¹).toReal ^ (1 / (finrank ℚ K : ℝ))
+  have hB : 0 ≤ B := Real.rpow_nonneg_of_nonneg toReal_nonneg _
+  have h_le : (minkowskiBound K) ≤ volume (convexBodySum K B) := by
+    refine le_of_eq ?_
+    rw [convexBodySum_volume, ← ENNReal.ofReal_pow hB, ← Real.rpow_nat_cast, ← Real.rpow_mul
+      toReal_nonneg, div_mul_cancel _ (Nat.cast_ne_zero.mpr (ne_of_gt finrank_pos)), Real.rpow_one,
+      ofReal_toReal, mul_comm, mul_assoc, ENNReal.inv_mul_cancel (convexBodySumFactor_ne_zero K)
+      (convexBodySumFactor_ne_top K), mul_one]
+    exact mul_ne_top (ne_of_lt (minkowskiBound_lt_top K))
+      (ENNReal.inv_ne_top.mpr (convexBodySumFactor_ne_zero K))
+  obtain ⟨x, h_nz, h_bd⟩ := exists_ne_zero_mem_ringOfIntegers_of_norm_le K h_le
+  refine ⟨x, h_nz, ?_⟩
+  convert h_bd
+  rw [div_pow B, ← Real.rpow_nat_cast B, ← Real.rpow_mul (by positivity), div_mul_cancel _
+    (Nat.cast_ne_zero.mpr <| ne_of_gt finrank_pos), Real.rpow_one, Nat.cast_pow, mul_comm_div,
+    mul_div_assoc']
+  congr 1
+  rw [eq_comm]
+  calc
+    _ = (2:ℝ)⁻¹ ^ NrComplexPlaces K * sqrt ‖discr K‖₊ * (2:ℝ) ^ finrank ℚ K *
+          ((2:ℝ) ^ NrRealPlaces K * (π / 2) ^ NrComplexPlaces K /
+          (Nat.factorial (finrank ℚ K)))⁻¹ := by
+      simp_rw [minkowskiBound, convexBodySumFactor, volume_fundamentalDomain_latticeBasis,
+        toReal_mul, toReal_inv, toReal_div, toReal_mul, coe_toReal, toReal_pow, toReal_inv,
+        toReal_ofNat, mixedEmbedding.finrank, NNReal.coe_pow, NNReal.coe_div, coe_real_pi,
+        NNReal.coe_ofNat, toReal_nat]
+    _ = (2:ℝ) ^ (finrank ℚ K - NrComplexPlaces K - NrRealPlaces K + NrComplexPlaces K : ℤ) *
+          Real.sqrt ‖discr K‖ * Nat.factorial (finrank ℚ K) * π⁻¹ ^ (NrComplexPlaces K) := by
+      simp_rw [inv_div, div_eq_mul_inv, mul_inv, ← zpow_neg_one, ← zpow_coe_nat, mul_zpow,
+        ← zpow_mul, neg_one_mul, mul_neg_one, neg_neg, Real.coe_sqrt, coe_nnnorm, sub_eq_add_neg,
+        zpow_add₀ (two_ne_zero : (2:ℝ) ≠ 0)]
+      ring
+    _ = (2:ℝ) ^ (2 * NrComplexPlaces K : ℤ) * Real.sqrt ‖discr K‖ * Nat.factorial (finrank ℚ K) *
+          π⁻¹ ^ (NrComplexPlaces K) := by
+      congr
+      rw [← card_add_two_mul_card_eq_rank, Nat.cast_add, Nat.cast_mul, Nat.cast_ofNat]
+      ring
+    _ = (4 / π) ^ NrComplexPlaces K * (finrank ℚ K).factorial * Real.sqrt |discr K| := by
+      rw [show ‖discr K‖ = |(discr K : ℝ)| by rfl, zpow_mul, show (2:ℝ) ^ (2:ℤ) = 4 by norm_cast,
+        div_pow, inv_eq_one_div, div_pow, one_pow, zpow_coe_nat]
+      ring
+
+theorem abs_discr_ge (h : 1 < finrank ℚ K) :
+    ((4:ℝ) / (9:ℝ)) * (3 * π / 4) ^ finrank ℚ K ≤ |discr K| := by
```

Comment:
```suggestion
    (4 / 9 : ℝ) * (3 * π / 4) ^ finrank ℚ K ≤ |discr K| := by
```
(untested)

Reply:
Yes, that works, thanks! I tried some other combinations that resulted in having some ↑ here and there, but this works fine.